lib/rules/no-lone-blocks.js

```diff
@@ -11,17 +13,94 @@
 
 module.exports = function(context) {
 
-    return {
-        "BlockStatement": function (node) {
-            // Check for any occurrence of BlockStatement > BlockStatement or
-            // Program > BlockStatement
-            var parent = context.getAncestors().pop();
-            if (parent.type === "BlockStatement" || parent.type === "Program") {
-                context.report(node, "Block is nested inside another block.");
+    // The lone blocks to be checked for block-level bindings
+    var loneBlocks = [],
+        ruleDef;
+
+    /**
+     * Reports a node as invalid.
+     * @param {ASTNode} node - The node to be reported.
+     * @returns {void}
+    */
+    function report(node) {
+        context.report(node, "Block is nested inside another block.");
+    }
+
+    /**
+     * Checks for any ocurrence of BlockStatement > BlockStatement or Program > BlockStatement
```

Comment:
I'm not sure why this was in the original code, but if the block's parent is `Program` then "Block is nested inside another block." is pretty strange message. @btmills any insights?